### ISSUE TITLE
Is executeVanillaTest broken?

### ISSUE THREAD

**Issue by justinpenner** [robotools/vanilla#211]
I can't get a simple test from the command line to work, at all.

```py
import vanilla
from vanilla.test.testTools import executeVanillaTest
print(vanilla.__version__)

class MyWindow:
    def __init__(self):
        self.w = vanilla.Window(
            (1000, 600),
            "Test",
        )
        self.w.open()

executeVanillaTest(MyWindow)
```

Output:
```
0.5.0
Traceback (most recent call last):
  File "./vanilla_test.py", line 13, in <module>
    executeVanillaTest(MyWindow)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/vanilla/test/testTools.py", line 59, in executeVanillaTest
    cls(**kwargs)
  File "./vanilla_test.py", line 11, in __init__
    self.w.open()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/vanilla/vanillaWindows.py", line 266, in open
    if self._window is None:
       ^^^^^^^^^^^^
AttributeError: 'Window' object has no attribute '_window'
```

So, it seems like when I instantiate `vanilla.Window`, its `__init__` method isn't getting called for some reason.

**Comment by justinpenner** [comment]
I just discovered that this problem is unique to pyobjc 10.3. If I downgrade to pyobjc 10.2 it resolves this issue and my test window opens:

<img width="1112" alt="image" src="https://github.com/robotools/vanilla/assets/46039243/0c0f8f0a-a4dc-4304-b1ba-8a3231fe274c">

Additionally, pyobjc 10.3 breaks vanilla when I use it in a Glyphs plugin (running vanilla 0.5.0), with the following errors:

![image](https://github.com/robotools/vanilla/assets/46039243/b11bc2e7-3717-49b3-aaf3-41a16701d989)

**Comment by justinpenner** [comment]
I'm nowhere near able to wrap my head around pyobjc yet, but I suspect the problem could be related to this change from the [10.3 release notes](https://github.com/ronaldoussoren/pyobjc/releases/tag/v10.3):

- Instantiating an Objective-C class by calling the class (e.g. invoking
`__new__`) will not call `__init__` even if one is defined.

**Comment by justvanrossum** [comment]
Your suspicion could be right, it may be this: https://github.com/ronaldoussoren/pyobjc/issues/610

**Comment by justvanrossum** [comment]
Yes, it is, vanilla.Window uses that exact patterns with `__new__` and `__init__`.

**Comment by justinpenner** [comment]
This is fixed as of pyobjc 10.3.1.
